src/main/java/org/opensearch/neuralsearch/processor/NormalizationProcessorWorkflow.java

```diff
@@ -276,14 +316,21 @@ private void updateOriginalFetchResults(
 
         QuerySearchResult querySearchResult = querySearchResults.get(0);
         TopDocs topDocs = querySearchResult.topDocs().topDocs;
+        // Scenario to handle when calculating the trimmed length of updated search hits
+        // When normalization process runs after fetch phase, then search hits already fetched. Therefore, use the from value sent in the
+        // search request to calculate the effective length of updated search hits array.
+        int trimmedLengthOfSearchHits = topDocs.scoreDocs.length - fromValueForSingleShard;
         // iterate over the normalized/combined scores, that solves (1) and (3)
-        SearchHit[] updatedSearchHitArray = Arrays.stream(topDocs.scoreDocs).map(scoreDoc -> {
+        SearchHit[] updatedSearchHitArray = new SearchHit[trimmedLengthOfSearchHits];
+        for (int i = fromValueForSingleShard; i < topDocs.scoreDocs.length; i++) {
+            // Read topDocs after the desired from length
+            ScoreDoc scoreDoc = topDocs.scoreDocs[i];
             // get fetched hit content by doc_id
             SearchHit searchHit = docIdToSearchHit.get(scoreDoc.doc);
             // update score to normalized/combined value (3)
             searchHit.score(scoreDoc.score);
-            return searchHit;
-        }).toArray(SearchHit[]::new);
+            updatedSearchHitArray[i - fromValueForSingleShard] = searchHit;
```

Comment:
we need to check if `i - fromValueForSingleShard` is a valid index, within [0, array.length- 1] boundaries

Reply:
Well, if you look at the code it is just putting values in the fresh array updateSearchHitArray. i starts from fromValueForSingleShard. So i-fromValueForSingleShard will be 0.  i can maximum go to array.length- 1 so it has already been taken care off. Do you still want me to put extra assert condition here?

Reply:
Understood, my intention is that we have array index iterator that goes from 0 to the length -1, we can adjust the logic for the for loop temp variable, something like this:
 
for (int i = 0; i < Math.min(topDocs.scoreDocs.length - topDocs.scoreDocs.length, updatedSearchHitArray.length); i++) {
     ScoreDoc scoreDoc = topDocs.scoreDocs[i + fromValueForSingleShard];
     ....
     updatedSearchHitArray[i] = searchHit;